From: Jean Delvare <jdelvare@suse.de>
Subject: Revert "dmi_scan: fix missing check for _DMI_ signature in smbios_present()"
Patch-mainline: Never

Revert 3.0.69 commit 3241d80d48900fc43535980865f1ef726af68713
(commit a40e7cf8f06b4e322ba902e4e9f6a6b0c2daa907 upstream.)

This is a fixup for a reverted patch so we must revert this one first.

Acked-by: Jean Delvare <jdelvare@suse.de>
---
 drivers/firmware/dmi_scan.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- a/drivers/firmware/dmi_scan.c
+++ b/drivers/firmware/dmi_scan.c
@@ -442,6 +442,7 @@ static int __init dmi_present(const char
 static int __init smbios_present(const char __iomem *p)
 {
 	u8 buf[32];
+	int offset = 0;
 
 	memcpy_fromio(buf, p, 32);
 	if ((buf[5] < 32) && dmi_checksum(buf, buf[5])) {
@@ -460,9 +461,9 @@ static int __init smbios_present(const c
 			dmi_ver = 0x0206;
 			break;
 		}
-		return memcmp(p + 16, "_DMI_", 5) || dmi_present(p + 16);
+		offset = 16;
 	}
-	return 1;
+	return dmi_present(buf + offset);
 }
 
 void __init dmi_scan_machine(void)
